jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBCCore.java

```diff
@@ -277,14 +279,45 @@ public synchronized void executePolling(String pollQuery) {
     * @param message        The Query message
     */
    public void sendDataFromQuery(HashMap[] queryArray, ExtensionServiceMessage message) {
+       Map<String, ?> request = (Map<String, ?>) message.getObject();
        String replyAddress = ExtensionServiceMessage.extractReplyAddress(message);
        
+       int bundleFactor = DEFAULT_BUNDLE_SIZE;
+       if (request.get("bundleFactor") instanceof Integer && (Integer) request.get("bundleFactor") > -1) {
+           bundleFactor = (Integer) request.get("bundleFactor");
+       }
+       
        // Send the results of the query
        if (queryArray.length == 0) {
            // If data is empty send empty map with 204 code
            client.sendQueryResponse(204, replyAddress, new LinkedHashMap<>());
-       } else {
+           lastRowBundle = null;
+       } else if (bundleFactor == 0) {
+           // If the bundleFactor was specified to be 0, then we sent the entire array
            client.sendQueryResponse(200, replyAddress, queryArray);
+           lastRowBundle = queryArray;
+       } else {
+           // Otherwise, send messages containing 'bundleFactor' number of rows
+           int len = queryArray.length;
+           
+           for (int i = 0; i < len - bundleFactor + 1; i += bundleFactor) {
+               HashMap[] rowBundle = Arrays.copyOfRange(queryArray, i, i + bundleFactor);
```

Comment:
Seems like it'd be more straightforward to say something like
```
int rowsRemaining = queryArray.length
while (rowsRemaining > 0) {
    int rowsToCopy = (bundleFactor <= rowsRemaining ? rowsRemaining : bundleFactor)
    ... Arrays.copyOf(queryArray, I, rowsToCopy)
    ...
    rowsRemaining -= rowsToCopy
    if (rowsRemaining <= 0) {
        // send 200
    } else {
       // send 100
    }
...
}
```
Then you can get rid of the "check for remainder thing"